### PR TITLE
mujoco: 2.3.0 -> 2.3.7

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/deepmind/mujoco/releases/tag/2.3.7

@samuela 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/robotics/mujoco/default.nix [hunk 1]
```diff
@@ -49,24 +49,24 @@ let
...
-    rev = "3df027b91202cf179f3fba3c46eebe65bbac3790";
-    hash = "sha256-aHO5n9Y35C7/zb3surfMyjyMjo109DoZnkozhiAKpYQ=";
+    rev = "0c8fc90d2037588024d9964515c1e684f6007ecc";
+    hash = "sha256-Ptzxad3ewmKJbbcmrBT+os4b4SR976zlCG9F0nq0x94=";
   };
   tinyobjloader = fetchFromGitHub {
     owner = "tinyobjloader";
     repo = "tinyobjloader";
     rev = "1421a10d6ed9742f5b2c1766d22faa6cfbc56248";
-    hash = "sha256-9z2Ne/WPCiXkQpT8Cun/pSGUwgClYH+kQ6Dx1JvW6w0=";
```
Comment (samuela) [inline pkgs/applications/science/robotics/mujoco/default.nix:59]: why this diff? if unnecessary, let's not change so as to keep the git blame clean

Comment (GaetanLepage) [reply pkgs/applications/science/robotics/mujoco/default.nix]: Oh yes, my mistake !

**Review by samuela — APPROVED** [review]
diff LGTM

**Comment by GaetanLepage** [comment]
> diff LGTM

Cool ! Do you have the rights to merge ?

**Comment by GaetanLepage** [comment]
> Cool ! Do you have the rights to merge ?

Thanks so much !

**Pull request merged**